lesson9

KochevniKXXI opened the pull request.

KochevniKXXI committed:
Девятый урок

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/MainActivity.kt:
```diff
@@ -0,0 +1,115 @@
+
+    private val progressReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent?) {
+            intent?.getIntExtra(DownloadService.KEY_PROGRESS, 0)?.let {
```
у меня загрука по дебагу идет в экстеншене, но на активити сюда уже не заходит

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/MainActivity.kt:
```diff
@@ -0,0 +1,115 @@
+            weatherService = (binder as WeatherService.WeatherBinder).service.apply {
+                serviceCallbacks = this@MainActivity
+            }
+            weatherService?.getWeather()
```
сервис при такой настройке будет жить пока он забинжен, получить покоду можно в onCreate сервиса вызвать

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/MainActivity.kt:
```diff
@@ -0,0 +1,115 @@
+    override fun onStart() {
+        super.onStart()
+        Intent(this, WeatherService::class.java).also { intent ->
+            bindService(intent, connection, BIND_AUTO_CREATE)
```
ты биндишь в onStart, анбиндишь в onDestory, нет симметрии, биндить нужно в onCreate

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/ThermometerView.kt:
```diff
@@ -0,0 +1,234 @@
+import android.util.AttributeSet
+import android.view.View
+
+class ThermometerView(
```
классно

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/ThermometerView.kt:
```diff
@@ -0,0 +1,234 @@
+
+    private val basePaint = Paint().apply { color = Color.GRAY }
+    private val baseRectangle = RectF()
+    private val basePadding = 10f
```
на уроке дойдем, как будто тут какие то размеры, и они должны браться из ресурсов через конверт dp в пиксели

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/services/DownloadService.kt:
```diff
@@ -0,0 +1,99 @@
+                }
+            }
+
+            sendBroadcast(intentProgress.apply {
```
вот до сюда доходит, но в активити вообще никак

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/services/DownloadService.kt:
```diff
@@ -0,0 +1,99 @@
+            }
+        )
+        CoroutineScope(Dispatchers.IO).launch {
+            withContext(Dispatchers.IO) {
```
у тебя скоуп и так на io, тут не обязательно менять, студию можно засапрессить было, потому что она хочет напрямую сменить поток на io, хотя он и так io

DmitriyTSt commented on Lesson_9_Zemskov/app/src/main/java/hh/school/lesson_9_zemskov/services/WeatherService.kt:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    fun getWeather() {
+        serviceScope.launch {
```
вот это вот в onCreate можно бахнуть

KochevniKXXI committed:
Девятый урок.

Получение погоды перенесено в onCreate WeatherService.
Удалена лишняя смена потока в DownloadService

KochevniKXXI committed:
Девятый урок.

Получение погоды перенесено в onCreate WeatherService.
Удалена лишняя смена потока в DownloadService.
Исправлена потеря данных при передаче в Activity.
Доработан ThermometerView.
